contracts/base/BaseLiquidityHandler.sol

```diff
@@ -0,0 +1,231 @@
+            }
+            if (delta.amount1() < 0) {
+                range.key.currency1.settle(poolManager, sender, uint256(int256(-delta.amount1())), claims);
+            }
+            if (delta.amount0() > 0) {
+                range.key.currency0.take(poolManager, address(this), uint256(int256(delta.amount0())), true);
+            }
+            if (delta.amount1() > 0) {
+                range.key.currency1.take(poolManager, address(this), uint256(int256(delta.amount1())), true);
+            }
+        }
+
+        {
+            positions[sender][range.toId()].liquidity += liquidityToAdd;
+
+            // collected fees are credited to the position OR zero'd out
+            delta.amount0() > 0 ? position.tokensOwed0 += uint128(delta.amount0()) : position.tokensOwed0 = 0;
+            delta.amount1() > 0 ? position.tokensOwed1 += uint128(delta.amount1()) : position.tokensOwed1 = 0;
+        }
+        return delta;
+    }
+
+    function handleDecreaseLiquidity(
+        address owner,
+        LiquidityRange calldata range,
+        uint256 liquidityToRemove,
+        bytes calldata hookData,
+        bool useClaims
+    ) external returns (BalanceDelta) {
+        (BalanceDelta delta, BalanceDelta feesAccrued) = poolManager.modifyLiquidity(
+            range.key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: range.tickLower,
+                tickUpper: range.tickUpper,
+                liquidityDelta: -int256(liquidityToRemove),
+                salt: range.key.hooks.getLiquiditySalt(owner)
+            }),
+            hookData
+        );
+
+        // take all tokens first
+        // do NOT take tokens directly to the owner because this contract might be holding fees
+        // that need to be paid out (position.tokensOwed)
+        if (delta.amount0() > 0) {
+            range.key.currency0.take(poolManager, address(this), uint128(delta.amount0()), true);
+        }
+        if (delta.amount1() > 0) {
+            range.key.currency1.take(poolManager, address(this), uint128(delta.amount1()), true);
+        }
+
+        uint128 token0Owed;
+        uint128 token1Owed;
+        {
+            Position storage position = positions[owner][range.toId()];
+            (token0Owed, token1Owed) = _updateFeeGrowth(range, position);
+
+            BalanceDelta principalDelta = delta - feesAccrued;
+            token0Owed += position.tokensOwed0 + uint128(principalDelta.amount0());
+            token1Owed += position.tokensOwed1 + uint128(principalDelta.amount1());
+
+            position.tokensOwed0 = 0;
+            position.tokensOwed1 = 0;
+            position.liquidity -= liquidityToRemove;
+        }
+        {
+            delta = toBalanceDelta(int128(token0Owed), int128(token1Owed));
+
+            // sending tokens to the owner
+            if (token0Owed > 0) range.key.currency0.send(poolManager, owner, token0Owed, useClaims);
```

Comment:
i personally dont think we need to support sending 6909 to user.

i do think this contract could keep them as 6909s though (& maybe just by default it should? i think it would be cheaper)